cmd/ncdu/ncdu.go

```diff
@@ -150,77 +152,91 @@ type dirPos struct {
 	offset int
 }
 
+// graphemeWidth returns the number of cells in rs.
+//
+// The original [runewidth.StringWidth] iterates through graphemes
+// and uses this same logic. To avoid iterating through graphemes
+// repeatedly, we separate that out into its own function.
+func graphemeWidth(rs []rune) (wd int) {
+	// copied/adapted from [runewidth.StringWidth]
```

Comment:
This is something I'm not sure about. Do I properly attribute this code fragment?

The original code is MIT licensed.

Reply:
@ncw

Reply:
I think this is a small enough bit of code that no-one is going to be bothered